app/views/shared/_image_thumbnail.html.erb

```diff
@@ -14,6 +14,7 @@
   # Lightbox stuff. obs_data comes from the matrix_box presenter,
   # link_type from view (for naming links)
   lightbox_id = is_set ? "observation-set" : SecureRandom.uuid
+  lightbox_url = (User.current&.image_size == "full_size") ? full_url : huge_url
   caption = image_caption_html(image_id, obs_data, link_type)
```

Comment:
This just falls back to "huge", where it's been. But i'd like to take anything from "medium" up. 
The problem is many users have a `null` size.
I could use a frozen array mapping values to urls, but is that the elegant move?
```ruby
{ "medium" => medium_url, "large" => large_url, "huge" => huge_url, "full_size" => full_url  }
```

Reply:
I mentioned this earlier, we really need to do a migration with this PR that resets everyone's preference to "huge".  That's the setting everyone's been stuck on for the past few years, regardless of what their (unused) preference claims.  We should make it so that it works the same after we deploy this PR, until they choose a different size preference.

Reply:
The migration sounds good to me. 
I'd like to make medium and large a possibility though. Is that hash/array the way?

Reply:
Is it just 
```ruby
  def change
    User.update_all(image_size: "huge")
  end
```

Reply:
Note that in the fixtures, `rolf` has his pref set to `medium`, which is also the default.
I could update the fixture, but it's also fine as is... I do test changing it to `full_size`.

In case js integration tests ever load real images, `medium` seems definitely preferable for testing.

Reply:
> The migration sounds good to me. I'd like to make medium and large a possibility though. Is that hash/array the way?

I answered this earlier, but it must've been missed.  Why not `Image.url(@user.image_size, image_id)`.

> `User.update_all(image_size: "huge")`

Yeah, that sounds right.

Reply:
You sound like you know the code.

LOL, I was really tired yesterday, huh? That solutions was in the same file, four lines away, wasn't it. 

So - If we're going to call @user.image_size, I think that db column needs a default in the migration, too. My server crashed when not finding an image_size.

Reply:
Actually: Will your suggestion incur an extra lookup of a value we already have? Or would it be cached?
I guess the advantage of my verbose map is that it uses the urls we've already pulled, above.

Reply:
Ah, ok. `Image.url` is just a path builder, but it does check `source_exists?`, which must have some cost.